src/sentry/api/serializers/models/user_notifications.py

```diff
@@ -1,35 +1,48 @@
 from collections import defaultdict
+from typing import Iterable
 
 from sentry.api.serializers import Serializer
-from sentry.models import UserOption
-from sentry.notifications.legacy_mappings import get_legacy_key_from_fine_tuning_key
-from sentry.notifications.types import FineTuningAPIKey
+from sentry.models.integration import ExternalProviders
+from sentry.models import NotificationSetting, UserOption
+from sentry.notifications.legacy_mappings import (
+    get_type_from_fine_tuning_key,
+    map_notification_settings_to_legacy,
+)
+from sentry.notifications.types import FineTuningAPIKey, NotificationScopeType
+
+
+def handle_legacy(notification_type: FineTuningAPIKey, users: Iterable) -> Iterable:
```

Comment:
I've been running into circular dependency problems because the functions are top-level in the file. What would be really cool is if the `User` model inherited a `User` or `UserType` placeholder type.